src/segments/haskell.go

```diff
@@ -0,0 +1,63 @@
+package segments
+
+import (
+	"oh-my-posh/environment"
+	"oh-my-posh/properties"
+)
+
+type Haskell struct {
+	language
+
+	StackGhc bool
+}
+
+const (
+	UseStackGhc properties.Property = "use_stack_ghc"
+)
+
+func (h *Haskell) Template() string {
+	return languageTemplate
+}
+
+func (h *Haskell) Init(props properties.Properties, env environment.Environment) {
+	ghcRegex := `(?P<version>((?P<major>[0-9]+).(?P<minor>[0-9]+).(?P<patch>[0-9]+)))`
+	ghcCmd := &cmd{
+		executable: "ghc",
+		args:       []string{"--numeric-version"},
+		regex:      ghcRegex,
+	}
+
+	stackCmd := &cmd{
+		executable: "stack",
+		args:       []string{"ghc", "--", "--numeric-version"},
+		regex:      ghcRegex,
+	}
+
+	h.language = language{
+		env:                env,
+		props:              props,
+		extensions:         []string{"*.hs", "*.lhs", "stack.yaml", "package.yaml", "*.cabal", "cabal.project"},
+		versionURLTemplate: "https://www.haskell.org/ghc/download_ghc_{{ .Major }}_{{ .Minor }}_{{ .Patch }}.html",
+	}
+
+	useStackGhc := h.props.GetString(UseStackGhc, "never")
+	if useStackGhc == "always" {
```

Comment:
would be cleaner if we transform this to a switch statement on `useStackGhc`:

```golang
switch useStackGhc {
    case "always":
        h.language.commands = []*cmd{stackCmd}
 		h.StackGhc = true
 	case "package":
        _, err := h.language.env.HasParentFilePath("stack.yaml")
 		if err == nil {
 			h.language.commands = []*cmd{stackCmd}
 			h.StackGhc = true
 		}
    default:
        h.language.commands = []*cmd{ghcCmd}
 	    h.StackGhc = false
}
```

Reply:
problem is for `package` I needed the command to be `ghcCmd` if  `stack.yaml` was not found. So instead of using a default case I just initially set the cmd in the language block hope thats okay?